arch/arm/src/kinetis/kinetis_pinirq.c

```diff
@@ -459,4 +475,56 @@ void kinetis_pinirqdisable(uint32_t pinset)
     }
 #endif /* HAVE_PORTINTS */
 }
+
+/****************************************************************************
+ * Name: kinetis_gpiosetevent
+ *
+ * Description:
+ *   Sets/clears GPIO based event and interrupt triggers.
+ *
+ * Input Parameters:
+ *  - pinset: gpio pin configuration
+ *  - rising/falling edge: enables
+ *  - event:  generate event when set
+ *  - func:   when non-NULL, generate interrupt
+ *  - arg:    Argument passed to the interrupt callback
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure indicating the
+ *   nature of the failure.
+ *
+ ****************************************************************************/
+
+int kinetis_gpiosetevent(uint32_t pinset, bool risingedge, bool fallingedge,
```

Comment:
But I can't see any place use this new function.

Reply:
Looks like a pure, useless wrapper function that only serves the PX4 project.  I have resisted all such changes in the past.

Even thought the function is not used, all builds will pay the overhead of the PX4-specific function.

Reply:
Yes the board is will be used on has not been up streamed yet. 

If the linker is set up correctly dead code will be removed.

Reply:
> If the linker is set up correctly dead code will be removed.

Why not put the function in a separate file?